packages/models-library/tests/test_utils_string_substitution.py

```diff
@@ -0,0 +1,174 @@
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+# pylint: disable=too-many-arguments
+
+import sys
+from pathlib import Path
+
+import pytest
+from models_library.utils.string_substitution import (
+    SubstitutionsDict,
+    TemplateText,
+    substitute_all_legacy_identifiers,
+    upgrade_identifier,
+)
+
+
+@pytest.mark.parametrize(
+    "legacy,expected",
+    [
+        (
+            "%%container_name.sym-server%%",
+            "OSPARC_ENVIRONMENT_CONTAINER_NAME_SYM_SERVER",
```

Comment:
also tow test cases here for `${SERVICE_VERSION}` -> `SERVICE_VERSION` and `${SIMCORE_REGISTRY}` -> `SIMCORE_REGISTRY`